Documentation/ceph-object-bucket-claim.md

```diff
@@ -0,0 +1,241 @@
+1. the `label` value shown is the name of the provisioner. In this example the provisioner name is aws-s3.io/bucket. This `label` is added to the `OBC`, `Secret`, `ConfigMap`, and `OB`. Note that the "/" in the provisioner name has been replaced by a "-" to conform to Kubernetes label values.
+1. the generated, unique `bucket name` for the new bucket.
+1. `objectReference` to the generated `OB`.
+1. `objectReference` to the generated `ConfigMap`.
+1. `objectReference` to the generated `Secret`.
+1. phases of bucket creation:
+    - _Pending_: the operator is processing the request
+    - _Bound_: the operator finished processing the request and linked the OBC and OB
+    - _Released_: the OB has been deleted, leaving the OBC unclaimed but unavailable.
+    - _Failed_: not currently set.
+
+##### Generated Secret
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ceph-bucket [1]
+  namespace: rook-ceph
+  finalizers:
+  - objectbucket.io/finalizer
+  labels:
+    bucket-provisioner: aws-s3.io-bucket
+  ownerReferences:
+  - name: ceph-bucket [2]
+    ...
+type: Opaque
+data:
+  ACCESS_KEY_ID: BASE64_ENCODED-1
+  SECRET_ACCESS_KEY: BASE64_ENCODED-2
+  ... [3]
+```
+1. `name` is same as name of the `OBC`
+1. `ownerReference` makes this secret a child of the originating OBC for clean up purposes.
+1. `ACCESS_KEY_ID` and `SECRET_ACCESS_KEY` are the only secret keys defined by the provisioner.
+```
+
+##### Generated ConfigMap
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ceph-bucket
+  namespace: rook-ceph
+  finalizers:
+  - objectbucket.io/finalizer
+  labels:
+    bucket-provisioner: aws-s3.io-bucket
+  ownerReferences:
+  - name: rook-ceph
+    ...
+data:
+  BUCKET_HOST: http://my-store
+  BUCKET_PORT: 80
+  BUCKET_NAME: photo-booth-62PrQ
+  BUCKET_REGION: us-west-1
+```
+
+##### App Pod
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: app-pod
+  namespace: dev-user
+spec:
+  containers:
+  - name: mycontainer
+    image: redis
+    envFrom: [1]
+    - configMapRef:
+        name: ceph-bucket [2]
+    - secretRef:
+        name: ceph-bucket [3]
+```
+1. use `env:` if mapping of the defined key names to the env var names used by the app is needed.
+1. makes available to the pod as env variables: `BUCKET_HOST`, `BUCKET_PORT`, `BUCKET_NAME`
+1. makes available to the pod as env variables: `ACCESS_KEY_ID`, `SECRET_ACCESS_KEY`
+
+##### Generated OB Custom Resource
+```yaml
+apiVersion: objectbucket.io/v1alpha1
+kind: ObjectBucket
+metadata:
+  creationTimestamp: "2019-10-02T01:35:36Z"
+  finalizers:
+  - objectbucket.io/finalizer
+  generation: 1
+  labels:
+    bucket-provisioner: aws-s3.io-bucket
+  name: obc-default-ceph-bucket
+  resourceVersion: "7761"
+  selfLink: /apis/objectbucket.io/v1alpha1/objectbuckets/obc-s3-provisioner-myobc
+  uid: eeb781ae-e4b4-11e9-ad72-982e45ddf979
+spec:
+  additionalState:
+    ARN: arn:aws:iam::939345161466:policy/photo-booth-62PrQ
+    UserName: photo-booth-62PrQ
```

Comment:
keys are usually camelCase. is this how they're actually generated?

Reply:
No, I just put a random name, in the version I changed this part